packages/flutter/lib/src/painting/_network_image_web.dart

```diff
@@ -80,18 +82,24 @@ class NetworkImage
     NetworkImage key,
     image_provider.DecoderCallback decode,
     StreamController<ImageChunkEvent> chunkEvents,
-  ) {
+  ) async {
     assert(key == this);
 
     final Uri resolved = Uri.base.resolve(key.url);
-    // This API only exists in the web engine implementation and is not
-    // contained in the analyzer summary for Flutter.
-    return ui.webOnlyInstantiateImageCodecFromUrl(// ignore: undefined_function, avoid_dynamic_calls
```

Comment:
We probably want to use `webOnlyInstantiateImageCodecFromUrl` when there are no HTTP headers specified. This would allow the engine to use the plain `<img>` element which is a better optimized method for rendering images. In particular `<img>` supports cross-origin images (at least for now; later it will depend on COOP/COEP trends).